source/stub/stub.go

```diff
@@ -72,14 +72,14 @@ func (s *Stub) Next(version uint) (nextVersion uint, err error) {
 
 func (s *Stub) ReadUp(version uint) (r io.ReadCloser, identifier string, err error) {
 	if m, ok := s.Migrations.Up(version); ok {
-		return ioutil.NopCloser(bytes.NewBufferString(m.Identifier)), fmt.Sprintf("%v.up.stub", version), nil
+		return ioutil.NopCloser(bytes.NewBufferString(m.Raw)), fmt.Sprintf("%v.up.stub", version), nil
 	}
 	return nil, "", &os.PathError{fmt.Sprintf("read up version %v", version), s.Url, os.ErrNotExist}
 }
 
 func (s *Stub) ReadDown(version uint) (r io.ReadCloser, identifier string, err error) {
 	if m, ok := s.Migrations.Down(version); ok {
-		return ioutil.NopCloser(bytes.NewBufferString(m.Identifier)), fmt.Sprintf("%v.down.stub", version), nil
+		return ioutil.NopCloser(bytes.NewBufferString(m.Raw)), fmt.Sprintf("%v.down.stub", version), nil
```

Comment:
Since `bodySequence` attempts to assert on the `Body` of the migration, put the full `Raw` data here instead of just an `Identifier`.

As far as I can tell, before this pull request `sourceStubMigrations` never filled in the `Identifier` field anyway, so this was always an empty string. Which by extension meant that the 3 tests `testMigrate', 'testSteps' and 'testUpAndDown` were really only asserting the length of each migration sequence, never the actual data read from each migration.